docs/docs/components/api/README.md

```diff
@@ -1,249 +1,57 @@
 # Component API Reference
```

Comment:
Biggest changes in the PR are here:

- Slightly re-organized the doc, e.g. previously we started with information about the component lifecycle, but since that's relevant only to sources, I moved that down.
- Added docs for component actions
- Used the correct terms in the right places ("components" where something applied to all components, or "sources" or "actions" where something is specific to one or the other).
- Other random fixes to links, etc.